### PR TITLE
feat: migrate scanner from C++ to C and additional bindings of C, go, python, swift

### PR DESCRIPTION
**Pull request by ilyakochik**
Migrating scanner to C as C++ scanners are being deprecated in tree-sitter:
```
tree-sitter-sql-bigquery % tree-sitter test
Warning: Using a C++ scanner is now deprecated. Please migrate your scanner code to C,
as C++ support will be removed in the near future.
```

As a side benefit, tree-sitter created bindings for Python and other languages.

All tests pass, except for syntax highlighting on `select.sql` and `table.sql`. This is likely due to weird behaviour of `tree-sitter test` that works differently to `tree-sitter highlight` (https://github.com/tree-sitter/tree-sitter/issues/3239)

### CODE REVIEW

**Comment by ilyakochik** [comment]
https://github.com/takegue/tree-sitter-sql-bigquery/pull/75/commits/1c29e40fde4285bb485f651c3e8ad074a27c5ced fixes the issues with highlight tests as suggested by the tree-sitter team in https://github.com/tree-sitter/tree-sitter/issues/3239

**Comment by ilyakochik** [comment]
@takegue , are you comfortable merging this?

**Comment by takegue** [comment]
@ilyakochik: sorry for late reply.
Thanks for Pull request and I feel this is a good suggestion

I'll take time to review this night. sorry for the wait

File: src/scanner.c [hunk 1]
```diff
@@ -0,0 +1,316 @@
+// The MIT License (MIT)
```
Comment (takegue) [inline src/scanner.c:1]: no diff with src/scannner.c

File: package.json [hunk 2]
```diff
@@ -11,25 +12,45 @@
...
+    "bindings/node/*",
+    "queries/*",
+    "src/**"
+  ],
   "repository": "https://github.com/takegue/tree-sitter-sql-bigquery",
   "author": "takegue <takegue@gmail.com>",
   "license": "MIT",
   "dependencies": {
-    "nan": "^2.14.2",
+    "tree-sitter-cli": "^0.20.6",
```
Comment (takegue) [inline package.json:37]: @ilyakochik: could you show me your tree-sitter-cli version?
I think some products in this PR are not reproduct-able from this version (i.e. src/tree-sitter/* files)

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "tree-sitter-sql-bigquery"
+description = "SqlBigquery grammar for tree-sitter"
+version = "0.0.1"
```
Comment (takegue) [inline pyproject.toml:8]:
```suggestion
version = "0.6.0"
```

File: Package.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                    "examples",
+                    ".editorconfig",
+                    ".github",
+                    ".gitignore",
+                    ".gitattributes",
+                    ".gitmodules",
+                ],
+                sources: [
+                    "src/parser.c",
+                    // NOTE: if your language has an external scanner, add it here.
```
Comment (takegue) [inline Package.swift:39]:
```suggestion
```

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        "tree_sitter_sql_bigquery.queries": ["*.scm"],
+    },
+    ext_package="tree_sitter_sql_bigquery",
+    ext_modules=[
+        Extension(
+            name="_binding",
+            sources=[
+                "bindings/python/tree_sitter_sql_bigquery/binding.c",
+                "src/parser.c",
+                # NOTE: if your language uses an external scanner, add it here.
```
Comment (takegue) [inline setup.py:39]:
```suggestion
```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,110 @@
+VERSION := 0.0.1
```
Comment (takegue) [inline Makefile:1]:
```suggestion
VERSION := 0.6.0
```

**Review by takegue — COMMENTED** [review]
reviewed

**Comment by takegue** [comment]
@ilyakochik: thanks for contribution and I have merged this PR.
I think additional works to release this change.
please wait the release. It will be soon

File: package.json (hunk 2 above)
Comment (ilyakochik) [reply package.json:37]: tree-sitter-cli@0.20.8

**Pull request merged**